ui/packages/catalog-ui-search/src/main/webapp/react-component/container/metacard-archive/metacard-archive.tsx

```diff
@@ -0,0 +1,163 @@
+import MetacardArchivePresentation from '../../presentation/metacard-archive'
+
+type Props = {
+  selectionInterface: any
+  model: any
+} & WithBackboneProps
+
+type State = {
+  collection: Backbone.Collection<Backbone.Model>
+  isDeleted: boolean
+  loading: boolean
+}
+
+class MetacardQuality extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    const selectionInterface = props.selectionInterface || store
+    const collection = props.model || selectionInterface.getSelectedResults()
+
+    const isDeleted = collection.some((result: any) => {
+      result.isDeleted()
+    })
+
+    this.state = {
+      collection,
+      isDeleted,
+      loading: true,
+    }
+  }
+
+  componentDidMount() {
+    this.setState({ loading: false })
+  }
+
+  handleArchive = () => {
+    const body = JSON.stringify(
+      this.state.collection.map((result: any) => {
+        return result.get('metacard').get('id')
+      })
+    )
+
+    const self = this
+    this.props.listenTo(
+      ConfirmationView.generateConfirmation({
+        prompt:
+          'Are you sure you want to archive?  Doing so will remove the item(s) from future search results.',
+        no: 'Cancel',
+        yes: 'Archive',
+      }),
+      'change:choice',
+      async function(confirmation: any) {
+        if (confirmation.get('choice')) {
+          self.setState({ loading: true })
+
+          const res = await fetch('./internal/metacards', {
+            method: 'DELETE',
+            body,
+          })
+          if (!res.ok) {
+            announcement.announce({
+              title: 'Unable to archive the selected item(s).',
+              message: 'Something went wrong.',
+              type: 'error',
+            })
+          }
+
+          self.state.collection.forEach(function(result) {
+            result
+              .get('metacard')
+              .get('properties')
+              .set('metacard-tags', ['deleted'])
+            result.trigger('refreshdata')
+          })
+          self.refreshResults()
+
+          setTimeout(() => {
+            self.setState({ isDeleted: true, loading: false })
+          }, 2000)
+        }
+      }.bind(this)
+    )
+  }
+
+  handleRestore = () => {
+    const self = this
+    this.props.listenTo(
+      ConfirmationView.generateConfirmation({
+        prompt:
+          'Are you sure you want to restore?  Doing so will include the item(s) in future search results.',
+        no: 'Cancel',
+        yes: 'Restore',
+      }),
+      'change:choice',
+      function(confirmation: any) {
+        if (confirmation.get('choice')) {
+          self.setState({ loading: true })
+
+          const promises = self.state.collection.map((result: any) => {
+            fetch(
+              `./internal/history/revert/${result
+                .get('metacard')
+                .get('properties')
+                .get('metacard.deleted.id')}/${result
```

Comment:
This is hard to read. Please pull 

```javascript
result
  .get('metacard')
  .get('properties')
  .get('metacard.deleted.id')`
``` 

and the version call into 2 separate constants.